### PR TITLE
PLAT-1253 | annotations editor

### PR DESCRIPTION
**Pull request by kvas-damian**

### CODE REVIEW

File: app/Http/Controllers/Api/PrivateApi/ReferenceColorsApiController.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?php namespace App\Http\Controllers\Api\PrivateApi;
+
+use App\Http\Controllers\Api\ApiController;
+use App\Http\Requests\Course\UpdateCourse;
```
Comment (rogatty) [inline app/Http/Controllers/Api/PrivateApi/ReferenceColorsApiController.php]: Unused.

File: app/Http/Controllers/Api/Transformers/ReferenceColorTransformer.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	{
+		$this->parent = $parent;
+	}
+
+	public function transform(ReferenceColor $referenceColor)
+	{
+		$data = [
+			'id'         => $referenceColor->id,
+			'color'    => $referenceColor->color,
+			'name'       => $referenceColor->name,
```
Comment (rogatty) [inline app/Http/Controllers/Api/Transformers/ReferenceColorTransformer.php]: Let's reformat both transformers.

File: database/migrations/2019_05_10_095726_add_note_to_annotations.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('annotations', function (Blueprint $table) {
+					$table->text('note')->nullable();
```
Comment (rogatty) [inline database/migrations/2019_05_10_095726_add_note_to_annotations.php]: Let's reformat both migrations.

File: resources/assets/js/admin/components/lists/PaginatedList.vue [hunk 4]
```diff
@@ -75,10 +75,6 @@ export default {
 			type: Object,
 			default: () => ({}),
 		},
-		dirty: {
```
Comment (rogatty) [inline resources/assets/js/admin/components/lists/PaginatedList.vue]: Was it unused?

Comment (kvas-damian) [reply resources/assets/js/admin/components/lists/PaginatedList.vue]: yes, it was used on the old annotations list

File: resources/assets/js/admin/components/slides/annotations/AnnotationParserTag.vue [hunk 5]
```diff
@@ -0,0 +1,141 @@
+<template>
+	<div class="annotationTagParser">
```
Comment (rogatty) [inline resources/assets/js/admin/components/slides/annotations/AnnotationParserTag.vue]: We don't use camelCase in class names.

File: resources/assets/js/admin/components/slides/annotations/AnnotationParserTag.vue [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+			</div>
+		</div>
+	</div>
+</template>
+
+<style lang="sass" scoped>
+	@import 'resources/assets/sass/styleguide/settings/colors'
+	@import 'resources/assets/sass/styleguide/settings/spacings'
+
+	.annotationTagParser
```
Comment (rogatty) [inline resources/assets/js/admin/components/slides/annotations/AnnotationParserTag.vue]: `.annotation-parser-tag`

File: resources/assets/js/admin/components/slides/annotations/AnnotationParserTag.vue [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+			selectedColor: null,
+			selectedIcon: null,
+			label: '',
+		};
+	},
+	computed: {
+		...mapState('referenceIcons', ['icons']),
+		...mapState('referenceColors', ['colors']),
+		content() {
+			return `{a:${this.selectedColor}:${this.annotationId}:${this.selectedIcon}${this.label}{a}`;
```
Comment (rogatty) [inline resources/assets/js/admin/components/slides/annotations/AnnotationParserTag.vue]: `}` is missing.

File: resources/assets/js/admin/components/slides/annotations/AnnotationParserTag.vue [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+
+		if (this.colors.length) {
+			this.selectedColor = this.colors[0].id;
+		}
+		if (this.icons.length) {
+			this.selectedIcon = this.icons[0].id;
+		}
+	},
+	methods: {
+		...mapActions('referenceColors', { setupReferenceColors: 'setup'	}),
```
Comment (rogatty) [inline resources/assets/js/admin/components/slides/annotations/AnnotationParserTag.vue]: Formatting.

File: resources/assets/js/admin/store/store.js [hunk 9]
```diff
@@ -22,6 +22,8 @@ import courseStructure from 'js/admin/store/modules/courseStructure';
 import groups from './modules/groups';
 import taxonomies from 'js/store/modules/taxonomies';
 import contentClassifier from 'js/store/modules/contentClassifier';
+import referenceColors from 'js/admin/store/modules/referenceColors';
+import referenceIcons from './modules/referenceIcons';
```
Comment (rogatty) [inline resources/assets/js/admin/store/store.js]:
```suggestion
import referenceIcons from 'js/admin/store/modules/referenceIcons';
```

File: resources/assets/js/admin/components/slides/annotations/AnnotationParserTag.vue [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+			>
+				<input
+					:id="`annotationParserTagColor-${color.id}`"
+					v-model="selectedColor"
+					class="is-checkradio"
+					type="radio"
+					:value="color.id"
+				>
+				<label :for="`annotationParserTagColor-${color.id}`">
+					<span class="o-annotationParserTag__option__color" :style="{ 'background-color': color.color }" />{{color.name}}
```
Comment (bognix) [inline resources/assets/js/admin/components/slides/annotations/AnnotationParserTag.vue]: not sure if this should be an organism, it doesn't contain any atoms, nor molecules

Comment (kvas-damian) [reply resources/assets/js/admin/components/slides/annotations/AnnotationParserTag.vue]: It's a part of the organism. I don't expect it to be reused in other places.

File: app/Http/Controllers/Api/PrivateApi/AnnotationsApiController.php [hunk 11]
```diff
@@ -57,20 +43,9 @@ public function put(UpdateAnnotation $request, $annotationId) {
...
-
-		$annotation->tags()->sync($tagIds);
-		$annotation->keywords()->delete();
-		$annotation->keywords()->saveMany($keywords);
-
+		$annotation->title = $request->title;
+		$annotation->description = $request->description;
+		$annotation->note = $request->note;
 		$annotation->save();
 		$annotation->searchable();
```
Comment (bognix) [inline app/Http/Controllers/Api/PrivateApi/AnnotationsApiController.php]: Do we need to call it? Shouldn't the update be picked up automatically by Searchable trait?

File: resources/assets/js/admin/components/slides/annotations/AnnotationsList.vue [hunk 12]
```diff
@@ -1,153 +1,56 @@
 <template>
-	<ul>
-		<li
-			v-for="(annotation, index) in serializeResponse(list)"
+	<wnl-paginated-sortable-table
```
Comment (bognix) [inline resources/assets/js/admin/components/slides/annotations/AnnotationsList.vue]: it's ok to remove the ability to preview the annotation on the annotations list?

File: resources/assets/js/admin/components/slides/annotations/AnnotationParserTag.vue [hunk 13]
```diff
@@ -0,0 +1,141 @@
...
+
+	.o-annotationParserTag
+		&__option
+			&__icon
+				color: $color-cadet-blue
+				margin: 0 $space-xxs
+			&__color
+				border-radius: 100%
+				display: inline-block
+				height: 21px
```
Comment (bognix) [inline resources/assets/js/admin/components/slides/annotations/AnnotationParserTag.vue]: sizes can't be taken from styleguide?

File: resources/assets/js/admin/components/slides/annotations/AnnotationParserTag.vue [hunk 14]
```diff
@@ -0,0 +1,141 @@
+<template>
```
Comment (bognix) [inline resources/assets/js/admin/components/slides/annotations/AnnotationParserTag.vue]: I know it's super simple component but maybe you can consider adding unit tests for it? It's easier to add next tests once component get any more complex if the very first test already exists.

File: resources/assets/js/admin/store/modules/referenceColors.js [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+
+// Actions
+const actions = {
+	async setup({ commit }) {
+		commit(types.SET_REFERENCE_COLORS_LOADING, true);
+		try {
+			const { data } = await axios.get(getApiUrl('reference_colors/all'));
+			commit(types.SET_REFERENCE_COLORS, data);
+		} catch (error) {
+			throw error;
```
Comment (bognix) [inline resources/assets/js/admin/store/modules/referenceColors.js]: you throw this error and it's not caught in the component. Can you catch it and add some error handling?

File: resources/assets/js/components/global/form/Tags.vue [hunk 16]
```diff
@@ -5,23 +5,9 @@
 				v-for="tag in tags"
 				:key="tag.id"
 				class="tag"
-				@click="removeTag(tag)"
```
Comment (bognix) [inline resources/assets/js/components/global/form/Tags.vue]: why do we want to remove the ability to CRU tags?

**Comment by bognix** [comment]
please update the `TagEditor.vue` file so it redirects to correct annotation URL instead of doing this hack with search phrase.

File: resources/assets/js/admin/components/slides/annotations/AnnotationsList.vue (hunk 12 above)
Comment (kvas-damian) [reply resources/assets/js/admin/components/slides/annotations/AnnotationsList.vue]: I think we agreed to use regular paginated list. We don't provide preview anywhere else. So I'd say, let's wait for feedback from admins.

File: resources/assets/js/components/global/form/Tags.vue (hunk 16 above)
Comment (kvas-damian) [reply resources/assets/js/components/global/form/Tags.vue]: Adam said if support for tags adds scope we can remove them. It's relatively easy to support them in a read-only state.

Comment (rogatty) [reply resources/assets/js/components/global/form/Tags.vue]: This is used only in the annotations editor. We've agreed that the tags are there only for information. Editorial team will copy (?) them as taxonomy terms and we'll remove the tags altogether.

File: app/Http/Controllers/Api/Transformers/ReferenceColorTransformer.php [hunk 17]
```diff
@@ -4,24 +4,28 @@
 namespace App\Http\Controllers\Api\Transformers;
 
 
-use App\Models\Keyword;
 use App\Http\Controllers\Api\ApiTransformer;
+use App\Models\Page;
```
Comment (rogatty) [inline app/Http/Controllers/Api/Transformers/ReferenceColorTransformer.php]: I don't think we need it here.

File: app/Http/Controllers/Api/Transformers/ReferenceIconTransformer.php [hunk 18]
```diff
@@ -0,0 +1,38 @@
+<?php
+
+
+namespace App\Http\Controllers\Api\Transformers;
+
+
+use App\Http\Controllers\Api\ApiTransformer;
+use App\Models\Page;
+use App\Models\ReferenceColor;
```
Comment (rogatty) [inline app/Http/Controllers/Api/Transformers/ReferenceIconTransformer.php]: I don't think we need these two.

File: resources/assets/js/components/global/form/Tags.vue (hunk 16 above)
Comment (bognix) [reply resources/assets/js/components/global/form/Tags.vue]: But then we need to do it either need to remove them using the DB query. Leaving it here will allow admins to do the clean up.

Comment (rogatty) [reply resources/assets/js/components/global/form/Tags.vue]: We'll do a tags clean up migration anyway.

File: resources/assets/js/admin/components/slides/annotations/AnnotationParserTag.vue [hunk 19]
```diff
@@ -116,16 +118,26 @@ export default {
...
+			if (this.colors.length) {
+				this.selectedColor = this.colors[0].id;
+			}
+			if (this.icons.length) {
+				this.selectedIcon = this.icons[0].id;
+			}
+		} catch (error) {
+			$wnl.logger.error(error);
+			this.addAutoDismissableAlert({
+				text: this.$i18n.t('ui.error.defaultErrorHandle'),
```
Comment (bognix) [inline resources/assets/js/admin/components/slides/annotations/AnnotationParserTag.vue]: `this.$t()` should be sufficient.

**Pull request merged**